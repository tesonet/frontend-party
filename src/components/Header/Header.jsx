import React from "react";
import { PropTypes } from "prop-types";
import { Navbar, NavbarBrand } from "reactstrap";
import logo from "../../assets/images/logo-dark.svg";
import icon from "../../assets/icons/ico-lock.svg";
import "./Header.scss";

const Header = ({ onLogout }) => (
    <Navbar>
        <NavbarBrand>
            <img src={logo} className="header-logo" />
        </NavbarBrand>
        <a className="logout-link" onClick={onLogout}>
            <img src={icon} />
            Logout
        </a>
    </Navbar>
);

export default Header;

Header.propTypes = {
    onLogout: PropTypes.func
};
