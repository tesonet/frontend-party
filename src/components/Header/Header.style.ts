import styled, { css } from 'styled-components'

export const headerHeight = '7rem'

const fixedHeaderStyle = css`
  position: fixed;
  top: 0;
  right: 0;
  bottom: ${headerHeight};
  left: 0;
`

export const HeaderWrapper = styled.header<{ fixed: boolean }>`
  height: ${headerHeight};
  padding: 0 1rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: static;
  background-color: ${({ theme }) => theme.colors.white};

  ${({ fixed }) => fixed && fixedHeaderStyle};
`
