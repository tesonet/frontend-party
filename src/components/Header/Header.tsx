import * as React from 'react';
import styled from 'styled-components';
import { themeGet } from '@styled-system/theme-get';

import * as DarkLogo from '@assets/logo-dark.png';
import { LogoutButton } from '@components/LogoutButton/LogoutButton';

interface HeaderProps {
  onLogout: () => void;
}

const HeaderWrapper = styled.div`
  width: 100%;
  min-height: 9.4rem;
  display: flex;
  justify-content: space-between;
  padding: 0 1.5rem;
  background-color: ${themeGet('colors.white')};
  align-items: center;
`;

const Image = styled.img`
  height: 3rem;
`;

export const Header = ({ onLogout }: HeaderProps) => {
  return (
    <HeaderWrapper>
      <Image src={DarkLogo} alt="Testio" />
      <LogoutButton onClick={onLogout}>Logout</LogoutButton>
    </HeaderWrapper>
  );
};
