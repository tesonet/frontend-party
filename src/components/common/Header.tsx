import React from 'react';
import Box from '../styled/Box';
import styled from 'styled-components';
import { ReactComponent as LogoSVG } from '../../assets/logotype-testio_black.svg';

const HeaderStyled = styled(Box)`
  height: 113px;
  display: flex;
  align-items: center;
`;

const Logo = styled(LogoSVG)`
  display: block;
  height: 30px;
  margin-left: 15px;
`;

export default function Header() {
  return (
    <HeaderStyled>
      <Logo alt="Tesonet" />
    </HeaderStyled>
  );
}
