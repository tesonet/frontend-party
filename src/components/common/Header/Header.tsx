import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import styles from './Header.module.scss';
import logo from './assets/logo-testio-small.png';
import logOut from './assets/logout.svg';
import { withRouter } from 'react-router-dom';
import { ROUTES } from '../../../routes';

interface Props extends RouteComponentProps<any> {}
const HeaderComponent: React.FC<Props> = ({ history }) => {
  const onLogOut = () => {
    localStorage.removeItem('auth-token');
    history.push(ROUTES.login);
  };
  return (
    <div className={styles.container}>
      <img data-test="header-logo" src={logo} alt="Testio" />
      <button
        data-test="logout-button"
        className={styles.logoutButton}
        onClick={onLogOut}
      >
        <img src={logOut} alt="logout" />
        Logout
      </button>
    </div>
  );
};

export const Header = withRouter(HeaderComponent);
