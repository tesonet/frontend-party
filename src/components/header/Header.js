import React from 'react';
import style from './Header.scss';
import Logo from '../logo/Logo';
import Logout from '../logout/Logout';

export default () => (
  <div className={`container ${style.header}`}>
    <div className="row">
      <div className="col">
        <Logo />
      </div>
      <div className="col">
        <Logout />
      </div>
    </div>
  </div>
);
