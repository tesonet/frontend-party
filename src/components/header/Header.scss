
.header {
  padding-top: 47px;
  height: 114px;
}
