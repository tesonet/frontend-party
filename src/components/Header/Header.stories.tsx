import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Header } from './Header';

const stories: any = storiesOf('components/Header', module);

stories.add('default', () => <Header onLogout={action('logout clicked')} />);
