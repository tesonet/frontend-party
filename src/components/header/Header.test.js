import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Enzyme, { render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from './Header';

Enzyme.configure({ adapter: new Adapter() });

test('Header renders without crashing', () => {
  shallow(<Header />);
});

test('Header renders correct message', () => {
  const e = render(<MemoryRouter><Header /></MemoryRouter>);
  expect(e.html()).toMatchSnapshot();
});
