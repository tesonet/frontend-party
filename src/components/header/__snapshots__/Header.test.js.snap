// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correct message 1`] = `"<div class=\\"row\\"><div class=\\"col\\"><div><img src=\\"\\" alt=\\"Testio Logo\\"></div></div><div class=\\"col\\"><div><a href=\\"/\\">logout</a></div></div></div>"`;
